src/Compilers/CSharp/Test/Symbol/Symbols/Source/EventTests.cs

```diff
@@ -325,6 +327,119 @@ class C
+        {
+            var source = @"
+public class A
+{
+    public event System.Action<dynamic> E1;
+}";
+            var libComp = CreateCompilationWithMscorlib(text: source).VerifyDiagnostics(
+                // (4,32): error CS1980: Cannot define a class or member that utilizes 'dynamic' because the compiler required type 'System.Runtime.CompilerServices.DynamicAttribute' cannot be found. Are you missing a reference?
+                //     public event System.Action<dynamic> E1;
+                Diagnostic(ErrorCode.ERR_DynamicAttributeMissing, "dynamic").WithArguments("System.Runtime.CompilerServices.DynamicAttribute").WithLocation(4, 32),
+                // (4,41): warning CS0067: The event 'A.E1' is never used
+                //     public event System.Action<dynamic> E1;
+                Diagnostic(ErrorCode.WRN_UnreferencedEvent, "E1").WithArguments("A.E1").WithLocation(4, 41));
+        }
+
+        [Fact]
+        public void EventWithDynamicAttribute_CannotCompileWithoutSystemCore()
+        {
+            var source = @"
+public class A
+{
+    public event System.Action<dynamic> E1 { add {} remove {} }
+}";
+            var libComp = CreateCompilationWithMscorlib(text: source).VerifyDiagnostics(
+                // (4,32): error CS1980: Cannot define a class or member that utilizes 'dynamic' because the compiler required type 'System.Runtime.CompilerServices.DynamicAttribute' cannot be found. Are you missing a reference?
+                //     public event System.Action<dynamic> E1 { add {} remove {} }
+                Diagnostic(ErrorCode.ERR_DynamicAttributeMissing, "dynamic").WithArguments("System.Runtime.CompilerServices.DynamicAttribute").WithLocation(4, 32));
+        }
+
+        [Fact]
+        public void EventWithDynamicAttribute_IsLoadedAsDynamic()
+        {
+            var libText = @"
+public class C {
+    public event System.Action<dynamic> E1;
+}";
+            var libComp = CreateCompilationWithMscorlib(text: libText, references: new[] { MscorlibRef, SystemCoreRef });
+            Assert.False(libComp.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error));
+
+            byte[] libImage;
+            using (var output = new MemoryStream())
+            {
+                libComp.Emit(output, options: new EmitOptions(metadataOnly: true));
+                libImage = output.ToArray();
+            }
+            var libAssemblyRef = MetadataReference.CreateFromImage(libImage.AsImmutableOrNull());
+
+            var source = @"
+class LambdaConsumer
+{
+    void M(C c)
+    {
+        c.E1 += f => f.Bogus();
+    }
+}";
+            // The compilation succeeds because 'f' gets properly treated as dynamic 
+            var comp = CreateCompilationWithMscorlib(text: source, references: new[] { MscorlibRef, SystemCoreRef, CSharpRef, libAssemblyRef });
+            Assert.False(comp.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error));
```

Comment:
Success of compilation is verified by compiling the code, running it and observing result. Use CompileAndVerify helper instead. Adjust the code so that it prints something to the console if it operates as expected, CompileAndVerify is able to verify output against a baseline.